src/engine/OIBusEngine.class.js

```diff
@@ -431,13 +414,36 @@ class OIBusEngine extends BaseEngine {
       }, {})
   }
 
+  async getCacheStatsForApis() {
+    // Get points APIs stats
+    const pointApis = Object.values(this.activeApis).filter((api) => api.canHandleValues)
+    const valuesCacheActions = pointApis.map((api) => api.getValueCacheStats())
+    const pointApisStats = Promise.all(valuesCacheActions)
+
+    // Get files APIs stats
+    const fileApis = Object.values(this.activeApis).filter((api) => api.canHandleFiles)
+    const fileCacheActions = fileApis.map((api) => api.getFileCacheStats())
+    const fileApisStats = Promise.all(fileCacheActions)
+
+    // Merge results
+    return [...pointApisStats, ...fileApisStats]
+  }
+
+  getCacheStatsForProtocols() {
+    const pointProtocols = Object.values(this.activeProtocols).filter((protocol) => protocol.handlesPoints)
+    return pointProtocols.map((protocol) => ({
+      name: protocol.dataSource.name,
+      count: protocol.addPointsCount || 0,
+    }))
+  }
+
   /**
    * Get status information.
    * @returns {object} - The status information
    */
   async getStatus() {
-    const apisCacheStats = await this.cache.getCacheStatsForApis()
-    const protocolsCacheStats = await this.cache.getCacheStatsForProtocols()
+    const apisCacheStats = await this.getCacheStatsForApis()
+    const protocolsCacheStats = this.getCacheStatsForProtocols()
```

Comment:
I have the feeling that it may be a good idea to extract the statistics into a separate class or service (maybe associated to the web server ?)
The engine should focus on managing the scan modes, etc
Moreover, the memory part of the stats are not related to OIBusEngine but to the process. I write it there but let's make another issue for that.

Reply:
Indeed, a separate service would be better.